File: src/main/java/mnm/mods/tabbychat/client/gui/component/config/GuiSettingString.java
package mnm.mods.tabbychat.client.gui.component.config;

import mnm.mods.tabbychat.util.config.Value;
import mnm.mods.tabbychat.client.gui.component.GuiText;
import mnm.mods.tabbychat.client.gui.component.config.GuiSetting.GuiSettingWrapped;

/**
 * A Gui input that wraps a {@link GuiText}.
 *
 * @author Matthew
 */
public class GuiSettingString extends GuiSettingWrapped<String, GuiText> {

    public GuiSettingString(Value<String> setting) {
        super(setting, new GuiText());
    }

}

File: src/main/kotlin/mnm/mods/tabbychat/client/gui/component/config/GuiSettingString.kt
package mnm.mods.tabbychat.client.gui.component.config

import mnm.mods.tabbychat.client.gui.component.GuiText
import kotlin.reflect.KMutableProperty0

/**
 * A Gui input that wraps a [GuiText].
 */
class GuiSettingString(setting: KMutableProperty0<String>) : GuiSetting.ValueSetting<String, GuiText>(setting, GuiText())
